diff --git a/crates/ruff_fmt/pkg/ruff_fmt.js b/crates/ruff_fmt/pkg/ruff_fmt.js
index abaccc1..724a315 100644
--- a/crates/ruff_fmt/pkg/ruff_fmt.js
+++ b/crates/ruff_fmt/pkg/ruff_fmt.js
@@ -199,7 +199,7 @@ function debugString(val) {
 }
 /**
 * @param {string} input
-* @param {Config | undefined} config
+* @param {Config | undefined} [config]
 * @returns {string}
 */
 export function format(input, config) {
@@ -504,8 +504,17 @@ async function __wbg_init(input) {
     }
     const imports = __wbg_get_imports();
 
-    if (typeof input === 'string' || (typeof Request === 'function' && input instanceof Request) || (typeof URL === 'function' && input instanceof URL)) {
-        input = fetch(input);
+    switch (true) {
+        case typeof input === 'string':
+            input = new URL(input, import.meta.url);
+        case input instanceof URL:
+            if (typeof __webpack_require__ !== 'function' && input.protocol === 'file:') {
+                const fs = await import('node:fs/promises');
+                input = fs.readFile(input);
+                break;
+            }
+        case typeof Request === 'function' && input instanceof Request:
+            input = fetch(input);
     }
 
     __wbg_init_memory(imports);
